src/App.js

```diff
@@ -1,26 +1,63 @@
 import { useState } from 'react';
-import Card from './Card';
+
+import CardList from './Card/CardList';
 import './index.css';
 
 const App = () => {
-  
-  const [card, setCard] = useState({
+  const [viewModeOnlyChecked, setViewModeOnlyChecked] = useState(false);
+
+  const [cards, setCards] = useState([{
+    id: 1,
     title: 'Core Java Volume I – Fundamentals',
-    description: 'Core Java Volume I – Fundamentals is a Java reference book (Best book for Java)that offers a detailed explanation of various features of Core Java, including exception handling, interfaces, and lambda expressions. ',
-  });
+    description: 'Java reference book. ',
+  },
+  {
+    id: 2,
+    title: 'Title2',
+    description: 'Description2. ',
+  },
+  {
+    id: 3,
+    title: 'Title3',
+    description: 'Description3. ',
+  },
+  {
+    id: 4,
+    title: 'Title4',
+    description: 'Description4. ',
+  },
+  {
+    id: 5,
+    title: 'Title5',
+    description: 'Description5. ',
+  },
+  {
+    id: 6,
+    title: 'Title6',
+    description: 'Description6. ',
+  },
+  {
+    id: 7,
+    title: 'Title7',
+    description: 'Description7. ',
+  }
+  ]);
+
+  const cardUpdateHandler = updatedCard => {
+    const index = cards.findIndex(card => card.id === updatedCard.id);
+    cards[index] = updatedCard;
+    setCards([...cards]);
```

Comment:
Need to create a copy of  cards array and then a copy of precise ement prior to updating it. Initial object should be immutable until update via hook function.